### PR TITLE
feat: Oracle cleanup

### PR DESCRIPTION
**Pull request by adamewozniak**
## Description

Removes tobintaxery and updates `genesis.go` with keeper functions

closes: #250

----

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added appropriate labels to the PR
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by toteki — APPROVED** [review]
Looks good - the PR is mostly removals so it didn't take as long to review as I though it would at first glance.

File: x/oracle/genesis.go [hunk 1]
```diff
@@ -9,12 +11,107 @@ import (
...
+	}
+
+	for _, av := range genState.AggregateExchangeRateVotes {
+		valAddr, err := sdk.ValAddressFromBech32(av.Voter)
+		if err != nil {
+			panic(err)
+		}
+
+		keeper.SetAggregateExchangeRateVote(ctx, valAddr, av)
+	}
```
Comment (toteki) [inline x/oracle/genesis.go:58]: Does this mean that genesis state contains initial prices, as a consequence of hard-coded votes from validator addresses?

File: proto/umee/oracle/v1beta1/genesis.proto [hunk 2]
```diff
@@ -32,12 +31,4 @@ message FeederDelegation {
...
-}
-
-// TobinTax defines an denom and tobin_tax pair used in
-// oracle module's genesis state
-message TobinTax {
-  string denom     = 1;
-  string tobin_tax = 2
-      [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec", (gogoproto.nullable) = false];
-}
+}
```
Comment (alexanderbez) [inline proto/umee/oracle/v1beta1/genesis.proto:34]: nit: newline.

You add the following to your VSCode config:

```json
"files.insertFinalNewline": true,
"files.trimFinalNewlines": true,
"files.eol": "\n"
```

and it'll do everything for you on save 👍

File: x/oracle/genesis.go (hunk 1 above)
Comment (alexanderbez) [reply x/oracle/genesis.go:58]: I think the idea is this happens on export/import network upgrades. This process is old and somewhat antiquated, in favor of live upgrades via `x/upgrade`. So TLDR; in practice this code will never run except in maybe some tests cases.

File: x/oracle/keeper/reward.go [hunk 3]
```diff
@@ -18,7 +18,7 @@ func (k Keeper) RewardBallotWinners(
 	ballotWinners map[string]types.Claim,
 ) {
 	rewardDenoms := make([]string, len(voteTargets)+1)
-	rewardDenoms[0] = types.MicroUmeeDenom
+	rewardDenoms[0] = types.UmeeDenom
```
Comment (alexanderbez) [inline x/oracle/keeper/reward.go:21]: this actually need to be the base denom, i.e. `uumee`.

Just a quick note/recap on denominations in the state-machine:

- The state-machine only deals with and knows about "base-level" denominations, e.g. `uumee`, `uatom`, `uosmo`, etc...In other words, these "base-level" denoms are not divisible, they cannot be broken down any further.
- Denomination metadata exists in `x/bank` and such metadata is used to let users and clients know how to deal with and display "higher-level" denominations, e.g. `1 umee = 10^6 uumee (1000000uumee)`.
- The `x/oracle` module gets prices for high-level denominations, e.g. `atom` and thus we will use the metadata to do the conversions.

File: x/oracle/types/asset.go [hunk 4]
```diff
@@ -6,16 +6,16 @@ import (
...
-	USDDenom        = "USD"
-	BlocksPerMinute = uint64(10)
-	BlocksPerHour   = BlocksPerMinute * 60
-	BlocksPerDay    = BlocksPerHour * 24
-	BlocksPerWeek   = BlocksPerDay * 7
-	BlocksPerMonth  = BlocksPerDay * 30
-	BlocksPerYear   = BlocksPerDay * 365
-	MicroUnit       = int64(1e6)
+	AtomDenom       string = "ATOM"
+	UmeeDenom       string = "UMEE"
```
Comment (alexanderbez) [inline x/oracle/types/asset.go:10]: IDK how these are used yet in the module, but if they're used in regards to actually moving funds around, they need to be the "base-level" denoms, e.g. `uumee` and `uatom`.

Also, I doubt we need `AtomDenom` at all.

**Review by alexanderbez — CHANGES_REQUESTED** [review]
LGTM! Just one quick note on the denoms 👍

File: x/oracle/genesis.go (hunk 1 above)
Comment (adamewozniak) [reply x/oracle/genesis.go:58]: It contains the initial exchange rate and votes for them, but the votes don't turn into exchange rates until after the ballot is counted -- the exchange rates will always be at least one vote period behind the votes

**Review by alexanderbez — APPROVED** [review]

**Pull request merged**